src/prefect/server/models/events.py

```diff
@@ -419,6 +419,73 @@ async def work_pool_status_event(
     )
 
 
+async def work_pool_updated_event(
+    session: AsyncSession,
+    work_pool: "ORMWorkPool",
+    changed_fields: Dict[
+        str, Dict[str, Any]
+    ],  # {"field_name": {"old": value, "new": value}}
+    occurred: DateTime,
+) -> Event:
+    """Create an event for work pool field updates (non-status)."""
+    return Event(
+        occurred=occurred,
+        event="prefect.work-pool.updated",
+        resource={
+            "prefect.resource.id": f"prefect.work-pool.{work_pool.id}",
+            "prefect.resource.name": work_pool.name,
+            "prefect.work-pool.type": work_pool.type,
+            "prefect.resource.role": "work-pool",
+        },
+        payload={
+            "changed_fields": list(changed_fields.keys()),
+            "changes": changed_fields,
```

Comment:
I'd prefer to use "update" where possible in the payload to coincide with the event name.
```suggestion
            "updated_fields": list(changed_fields.keys()),
            "updates": changed_fields,
```